lib/lte_link_control/lte_lc.c

```diff
@@ -341,17 +405,28 @@ static void at_handler_cedrxp(const char *response)
 {
 	int err;
 	struct lte_lc_evt evt = {0};
+	char edrx_value[LTE_LC_EDRX_VALUE_LEN] = {0};
+	char ptw_value[LTE_LC_EDRX_VALUE_LEN] = {0};
 
 	__ASSERT_NO_MSG(response != NULL);
 
 	LOG_DBG("+CEDRXP notification");
 
-	err = parse_edrx(response, &evt.edrx_cfg);
+	err = parse_edrx(response, &evt.edrx_cfg, edrx_value, ptw_value);
 	if (err) {
 		LOG_ERR("Can't parse eDRX, error: %d", err);
 		return;
 	}
 
+	/* PTW must be requested after eDRX is enabled */
+	lte_lc_edrx_values_store(evt.edrx_cfg.mode, edrx_value, ptw_value);
+	/* Send PTW setting if eDRX is enabled, i.e., we have network mode */
+	if (evt.edrx_cfg.mode != LTE_LC_LTE_MODE_NONE) {
+		edrx_enabled = true;
```

Comment:
I thought that `edrx_enabled` would be set based on whether eDRX was requested by the application (`lte_lc_psm_req()`), not what the network has granted. Modem could be powered off in a cell which does not support eDRX, so `edrx_enabled` would be `false`. If modem is then powered on and camps on a cell which does support eDRX, there would be no unsolicited `+CEDRXP` notification even though eDRX would be used.

Reply:
True. Should be fixed now.